azure/Guidelines.md

```diff
@@ -747,18 +747,70 @@ The ability to retry failed requests for which a client never received a respons
+
+:white_check_mark: **DO** return a `200-OK` response, `201-Created` for create operations, from the request that initiates the operation.  The response body should contain a representation of the resource that clearly indicates that the operation has been accepted or started.
+
+:white_check_mark: **DO** support a get method on the resource that returns a representation of the resource including the status field that indicates when the operation has completed.
+
+:white_check_mark: **DO** define the "status" field of the resource as an enum with all the values it may contain including the "terminal" values "Succeeded", "Failed", and "Canceled".
+
+:ballot_box_with_check: **YOU SHOULD** use the name `status` for the "status" field of the resource.
+
+#### Stepwise long-running operations
+
+In a stepwise long-running operation, the client makes a request to initiate the operation processing and receives a URL in the response where it can obtain the operation results. The [HTTP specification](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.3) calls the target of this URL a "status monitor" but we will refer to it as the operation-results.
+
+:white_check_mark: **DO** return a `202-Accepted` status code from the request that initiates a stepwise LRO if the processing of the operation was successfully initiated.
+
+:white_check_mark: **DO** perform as much validation of the initial request as practical and return an error response when appropriate.
+
+:white_check_mark: **DO** return the operation-results URL in the `Operation-Location` response header.
+
+:white_check_mark: **DO** support the `get` method on the operation-results endpoint that returns a `200-OK` response with a response body that indicates whether the operation has completed and completion status (success, failed) with sufficient error information to diagnose failures.
+
+:white_check_mark: **DO** include a field in the operation-results named `status` to contain the operation status. This field should be a string with well defined values that must include "Succeeded", "Failed", and "Canceled".
+
+:white_check_mark: **DO** include a field in the operation-results named `error` to contain error information -- minimally `code` and `message` fields -- when an operation fails.
+
+:white_check_mark: **DO** support return of the operation-results for some documented period of time, a minimum of 24 hours, after the operation has completed.
+
+:white_check_mark: **DO** include a `Retry-After` header in the response to the initiating request and requests to the operation-results endpoint. The value of this header should be an integer number of seconds to wait before making the next request to the operations-results URL.
+
+:heavy_check_mark: **YOU MAY** support a `get` method on the operation-results collection endpoint that returns a list of all recently initiated operations.
```

Comment:
```suggestion
:heavy_check_mark: **YOU MAY** support a `get` method on the operation-results collection URL that returns a list of all recently initiated operations.
```

Reply:
"Recently initiated" is ambiguous. Does them in-flight?
I think we want to say a GET on the collection by default returns in-flight operations (which is probably a small set so a good default). And then, maybe we have some query parameters to further include completed, error, canceled, and maybe a timestamp range?

Reply:
I agree we probably should be more specific.  But I would go the other way -- GET returns all operations by default -- in-flight, completed (success or error), and cancelled, and then we can offer query parameters to select specific types or timestamp ranges.